subsys/bluetooth/mesh/access.c

```diff
@@ -1030,15 +1044,15 @@ static struct bt_mesh_model *bt_mesh_elem_find_group(struct bt_mesh_elem *elem,
 	return NULL;
 }
 
-struct bt_mesh_elem *bt_mesh_elem_find(uint16_t addr)
+const struct bt_mesh_elem *bt_mesh_elem_find(uint16_t addr)
 {
 	uint16_t index;
 
 	if (!BT_MESH_ADDR_IS_UNICAST(addr)) {
 		return NULL;
 	}
 
-	index = addr - dev_comp->elem[0].addr;
+	index = addr - dev_primary_addr;
```

Comment:
if `addr` is less than `dev_primary_addr` then the function returns a fake address.